code/modules/jobs/job_types/_job.dm

```diff
@@ -191,6 +191,11 @@
 		else
 			back = backpack //Department backpack
 
+	if(H.jumpsuit_style == PREF_SKIRT)
+		var/holder = addtext("[uniform]", "/skirt")
+		if(text2path(holder))	//if our uniform has a skirt variant as a child, and we have it set in preferences, set that as our uniform
+			uniform = holder
```

Comment:
Should probably set this to the result of `text2path` and not just as the text

Reply:
I'll remove the text2path on equip for uniform and put it here if it fails the if(text2path) check too, see if that breaks anything. If not I'll put up the change.